piparty.py

```diff
@@ -1009,9 +1009,9 @@ def ready_move(self, move):
     # Moves that are available to start a game, use force_all_start = False if just wanting to see alive moves
     def get_ready_moves(self, force_all_start):
         if force_all_start:
-            return [move.get_serial() for move in self.moves if self.out_moves.get(move.get_serial(), Status.DEAD.value) == Status.ALIVE.value and (self.menu_opts[move.get_serial()])[Opts.RANDOM_START.value]  ]
+            return [move.get_serial() for move in self.moves if self.out_moves.get(move.get_serial(), Status.DEAD.value) == Status.ALIVE.value and move.get_serial() in self.menu_opts and (self.menu_opts[move.get_serial()])[Opts.RANDOM_START.value]  ]
```

Comment:
Was getting a couple errors here:
2024-03-10 15:55:43,586 - DEBUG - controller_process - Tracking Move (Menu): 00:06:f7:22:98:27
    piparty.game_loop()
  File "/home/aaron/JoustMania/piparty.py", line 701, in game_loop
    self.update_status('menu')
  File "/home/aaron/JoustMania/piparty.py", line 922, in update_status
    'ready_count' : len(self.get_ready_moves(True)),
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aaron/JoustMania/piparty.py", line 1012, in get_ready_moves
    return [move.get_serial() for move in self.moves if self.out_moves.get(move.get_serial(), Status.DEAD.value) == Status.ALIVE.value and (self.menu_opts[move.get_serial()])[Opts.RANDOM_START.value]  ]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aaron/JoustMania/piparty.py", line 1012, in <listcomp>
    return [move.get_serial() for move in self.moves if self.out_moves.get(move.get_serial(), Status.DEAD.value) == Status.ALIVE.value and (self.menu_opts[move.get_serial()])[Opts.RANDOM_START.value]  ]
                                                                                                                                            ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
KeyError: None

I think this was happening when one of the controllers was plugged in and charging, when a game ended. This is a simple fix to prevent this

Reply:
updated in the latest commit, checking if move.get_serial() is in self.menu_opts
This happens when a controller is plugged in sometimes after a game ends